src/mame/drivers/dfs500.cpp

```diff
@@ -0,0 +1,1036 @@
+	// This goes to the CN21 EDITOR D-SUB CONNECTOR on board CN-573
+	// I think the purpose of this is to connect to an editor such as the Sony PVE-500.
+	//
+	// m_serial2->txd_handler().set(m_..., FUNC(..._device::write_txd)); "XMIT"
+	// m_...->rxd_handler().set(m_serial2, FUNC(i8251_device::write_rxd)); "RCV"
+
+	GENERIC_LATCH_16(config, "controlsignals");
+	GENERIC_LATCH_16(config, "reg7_5");
+	GENERIC_LATCH_16(config, "xflt");
+	GENERIC_LATCH_16(config, "yflt");
+
+	// In the future this could become IMAGE_AVIVIDEO (or even, perhaps, we
+	// should add support for capturing frames from real video devices such
+	// as a webcam on /dev/video0)
+	IMAGE_PICTURE(config, m_input1);
+	IMAGE_PICTURE(config, m_input2);
+	IMAGE_PICTURE(config, m_input3);
+	IMAGE_PICTURE(config, m_input4);
+
+	SCREEN(config, m_screen, SCREEN_TYPE_RASTER);
+	m_screen->set_refresh_hz(60);
+	m_screen->set_size(VIDEO_WIDTH, VIDEO_HEIGHT);
+	m_screen->set_visarea(0, VIDEO_WIDTH-1, 0, VIDEO_HEIGHT-1);
+	m_screen->set_screen_update(FUNC(dfs500_state::screen_update));
+
+	config.set_default_layout(layout_dfs500);
+}
+
+ROM_START(dfs500)
+	// Process Unit System Control:
+	ROM_REGION(0x8000, "cpanelcpu", 0)
+	ROM_LOAD("27c256b_npky14_v1.03_293-83_5500_ky223_sony94.ic14", 0x0000, 0x8000, CRC(8b9e564a) SHA1(aa8a1f211a7834fb15f7ecbc58570f566c0ef5ab))
+
+	// Process Unit System Control:
+	ROM_REGION(0x10000, "maincpu", 0)
+	ROM_LOAD16_BYTE("27c256b_npsys1_v1.03_293-84_3eb5_sy172_sony94.ic1", 0x0001, 0x8000, CRC(4604e7c0) SHA1(80f965b69a163a6278d6f54db741f4c5ada1cb59))
+	ROM_LOAD16_BYTE("27c256b_npsys2_v1.03_293-85_3ecd_sy172_sony94.ic2", 0x0000, 0x8000, CRC(b80a66e6) SHA1(407ddc5fee61920bfbe90c20faf4482ceef1ad4f))
+
+	// Process Unit Effect Control:
+	ROM_REGION(0x10000, "effectcpu", 0)
+	ROM_LOAD16_BYTE("27c512_npsys3_v1.04_293-86_b7d0_sy172_sony94.ic3", 0x0001, 0x8000, CRC(69238d02) SHA1(288babc7547858a3ca3f65af0be76f72335392ea))
+	ROM_CONTINUE(0x0001, 0x8000)
+	ROM_LOAD16_BYTE("27c512_npsys4_v1.04_293-87_b771_sy172_sony94.ic4", 0x0000, 0x8000, CRC(541abd4f) SHA1(e51f5ca6416c17535f2d2a13a7bedfb3b4b4a58b))
+	ROM_CONTINUE(0x0000, 0x8000)
+
+	// Process Unit Effect Data:
+	ROM_REGION(0x200000, "effectdata", 0)
+	ROM_LOAD("27c4001-12f1_sy172_v1.01_d216.ic5", 0x000000,  0x80000, CRC(ae094fcb) SHA1(c29c27b3c80e67caba2078bb60696c1b8692eb8b))
+	ROM_LOAD("27c4001-12f1_sy172_v1.01_d225.ic6", 0x080000,  0x80000, CRC(caa6ccb2) SHA1(9b72dc47cf4cc9c2f9915ea4f1bd7b5136e29db5))
+	ROM_LOAD("27c4001-12f1_sy172_v1.01_cc13.ic7", 0x100000,  0x80000, CRC(e1fe8606) SHA1(a573c7023daeb84d5a1182db4051b1bccfcfc1f8))
+	ROM_LOAD("27c4001-12f1_sy172_v1.01_c42d.ic8", 0x180000,  0x80000, CRC(66e0f20f) SHA1(e82562ae1eeecc5c97b0f40e01102c2ebe0d6276))
+ROM_END
+
+//   YEAR  NAME   PARENT/COMPAT MACHINE  INPUT    CLASS          INIT     COMPANY  FULLNAME                     FLAGS
+SYST(1994, dfs500,    0, 0,     dfs500, dfs500, dfs500_state, empty_init, "Sony", "DFS-500 DME Video Switcher", MACHINE_IS_SKELETON)
```

Comment:
The driver has considerably more information in it than what is typically thought of as a skeleton driver. Consider adjusting to `MACHINE_NOT_WORKING`.

Reply:
ok, great! I'll do it